test/unit/key_def.cc

```diff
@@ -457,16 +460,1330 @@ test_tuple_validate_key_parts_raw(void)
+		/* regular cases. */
+		test_key_new("[%u%u%u]", 0, 0, 1),
+		test_key_new("[%u%u%u]", 0, 0, 0),
+
+		test_key_new("[%u%u%u]", 0, 1, 0),
+		test_key_new("[%u%u%u]", 0, 0, 0),
+
+		test_key_new("[%u%u%u]", 0, 1, 0),
+		test_key_new("[%u%u%u]", 0, 0, 1),
+
+		test_key_new("[%u%u%u]", 0, 1, 1),
+		test_key_new("[%u%u%u]", 0, 0, 1),
+
+		test_key_new("[%u%u%u]", 1, 0, 0),
+		test_key_new("[%u%u%u]", 0, 0, 0),
+
+		test_key_new("[%u%u%u]", 1, 0, 0),
+		test_key_new("[%u%u%u]", 0, 0, 1),
+
+		test_key_new("[%u%u%u]", 1, 0, 0),
+		test_key_new("[%u%u%u]", 0, 1, 1),
+
+		test_key_new("[%u%u%u]", 1, 1, 1),
+		test_key_new("[%u%u%u]", 0, 1, 1),
+	};
+
+	if (is_nullable) {
+		/* NILs. */
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 0, 0));
+		keys_gt.push_back(test_key_new("[%u%uNIL]", 0, 0));
+
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 0, 0));
+		keys_gt.push_back(test_key_new("[%uNILNIL]", 0));
+
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 0, 0));
+		keys_gt.push_back(test_key_new("[NILNILNIL]"));
+
+		keys_gt.push_back(test_key_new("[%u%uNIL]", 0, 0));
+		keys_gt.push_back(test_key_new("[%uNILNIL]", 0));
+
+		keys_gt.push_back(test_key_new("[%u%uNIL]", 0, 0));
+		keys_gt.push_back(test_key_new("[NILNILNIL]"));
+
+		keys_gt.push_back(test_key_new("[%uNILNIL]", 0));
+		keys_gt.push_back(test_key_new("[NILNILNIL]"));
+
+		keys_gt.push_back(test_key_new("[%u%uNIL]", 0, 1));
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 0, 0));
+
+		keys_gt.push_back(test_key_new("[%uNILNIL]", 1));
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 0, 0));
+
+		keys_gt.push_back(test_key_new("[%uNILNIL]", 1));
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 1, 1));
+
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 0, 0));
+		keys_gt.push_back(test_key_new("[%uNIL%u]", 0, 1));
+
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 0, 0));
+		keys_gt.push_back(test_key_new("[NILNIL%u]", 1));
+
+		keys_gt.push_back(test_key_new("[%u%u%u]", 0, 0, 0));
+		keys_gt.push_back(test_key_new("[NIL%u%u]", 1, 1));
+
+		keys_gt.push_back(test_key_new("[%uNILNIL]", 1));
+		keys_gt.push_back(test_key_new("[%u%uNIL]", 0, 0));
+	}
+
+	fail_unless(keys_eq.size() % 2 == 0);
+	for (size_t i = 0; i < keys_eq.size(); i += 2) {
+		char *a = keys_eq[i];
+		char *b = keys_eq[i + 1];
+		test_key_compare_result(a, b, key_def, 0, funcname);
+		test_key_compare_result(b, a, key_def, 0, funcname);
+	}
+
+	fail_unless(keys_gt.size() % 2 == 0);
+	for (size_t i = 0; i < keys_gt.size(); i += 2) {
+		char *a = keys_gt[i];
+		char *b = keys_gt[i + 1];
+		test_key_compare_result(a, b, key_def, 1, funcname);
+		test_key_compare_result(b, a, key_def, -1, funcname);
+	}
+
+	for (size_t i = 0; i < keys_eq.size(); i++) {
+		free(keys_eq[i]);
+	}
+
+	for (size_t i = 0; i < keys_gt.size(); i++) {
+		free(keys_gt[i]);
+	}
+
+	key_def_delete(key_def);
+
+	free((void *)funcname);
+
+	footer();
+	check_plan();
+}
+
+static void
+test_tuple_compare_with_key(struct tuple *tuple_a, struct tuple *tuple_b,
```

Comment:
Nit: it would be nice to call such functions like `test_check_tuple_compare_with_key` to easily distinguish a test case from the function that checks a function.

Reply:
Renamed.